proposed/apphost-embed-install-location.md

```diff
@@ -0,0 +1,158 @@
+the .NET install location.
+
+Goals:
+
+- Enable developers to build an app and configure it such that it can:
+    - Only consider global .NET installs
+    - Always look at a relative path for a .NET install
+- Create a building block for a fuller, SDK-supported experience that should fit
+in regardless of where we land for the full experience
+
+Non-goals:
+
+- SDK support for layout and deployment of the app with a runtime files in a
+corresponding relative path
+- SDK and project system experience for building a set of applications using the
+same custom runtime install location
+
+Related:
+
+- [dotnet/runtime#2572](https://github.com/dotnet/runtime/issues/2572)
+- [dotnet/runtime#3453](https://github.com/dotnet/runtime/issues/3453)
+- [dotnet/runtime#53834](https://github.com/dotnet/runtime/issues/53834)
+- [dotnet/runtime#64430](https://github.com/dotnet/runtime/issues/64430)
+- [dotnet/runtime#70975](https://github.com/dotnet/runtime/issues/70975)
+- [dotnet/runtime#86801](https://github.com/dotnet/runtime/issues/86801)
+
+## State in .NET 8
+
+The current state is described in detail in
+[install-locations](https://github.com/dotnet/designs/blob/main/accepted/2020/install-locations.md)
+and [install-locations-per-architecture](https://github.com/dotnet/designs/blob/main/accepted/2021/install-locations-per-architecture.md)
+
+At a high level, the process and priority for `apphost` determining the install
+location is:
+
+  1. App-local
+      - Look for the runtime in the app's folder (self-contained apps)
+  2. Environment variables
+      - Read the `DOTNET_ROOT_<arch>` and `DOTNET_ROOT` environment variables
+  3. Global install (registered)
+      - Check for a registered install - registry key on Windows or a install
+      location file on non-Windows
+  4. Global install (default)
+      - Fall back to a well-known default install location based on the platform
+
+## Embedded install location options for `apphost`
+
+Every `apphost` already has a placeholder that gets rewritten to contain the app
+binary path during build (that is, `dotnet build`). This proposal adds another
+placeholder which would represent the optional configuration of search locations
+and embedded relative path to an install location. Like the existing placeholder
+placeholder, it would conditionally be rewritten during build based on the app's
+settings. This is similar to the proposal in [dotnet/runtime#64430](https://github.com/dotnet/runtime/issues/64430),
+but with additional configuration of which search locations to use.
+
+### Configuration of search locations
+
+Some applications do not want to use all the default search locations when they
+are deployed - for example, [dotnet/runtime#86801](https://github.com/dotnet/runtime/issues/86801).
+
+We can allow selection of which search locations the `apphost` will use. When
+search locations are configured, only the specified locations will be searched.
+
+```c++
```

Comment:
If this is C++, we should consider `enum class`. However, that gets into issues with scoping and flags with `enum class` are cumbersome without overloads such as `|` and `&`. As a result of that we should have a prefix for each of the values. For example,

```c
enum search_location
{
    sl_default = 0,
    sl_app_local = 1 << 0,
    sl_embedded = 1 << 1,
    sl_environment_variables = 1 << 2,
    sl_global = 1 << 3,
};
```